newrelic/resource_newrelic_obfuscation_expression.go

```diff
@@ -0,0 +1,200 @@
+				Required:    true,
+			},
+			"regex": {
+				Type:        schema.TypeString,
+				Description: "Regex of expression.",
+				Required:    true,
+			},
+			"expression_id": {
+				Type:        schema.TypeInt,
+				Description: "Expression Id.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+//Create the obfuscation expression
+func resourceNewRelicObfuscationExpressionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+
+	accountID := selectAccountID(providerConfig, d)
+
+	createInput := logconfigurations.LogConfigurationsCreateObfuscationExpressionInput{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Regex:       d.Get("regex").(string),
+	}
+
+	created, err := client.Logconfigurations.LogConfigurationsCreateObfuscationExpressionWithContext(ctx, accountID, createInput)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if created == nil {
+		return diag.Errorf("err: obfuscation expression create result wasn't returned or expression was not created.")
+	}
+
+	expressionID := created.ID
+
+	d.SetId(expressionID)
+
+	return resourceNewRelicObfuscationExpressionRead(ctx, d, meta)
+}
+
+//Read the obfuscation expression
+func resourceNewRelicObfuscationExpressionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+
+	expressionID := d.Id()
+	expression, err := getObfuscationExpressionByID(ctx, client, accountID, expressionID)
+
+	if err != nil {
+		if _, ok := err.(*nrErrors.NotFound); ok {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("account_id", accountID); err != nil {
+		return diag.FromErr(err)
+	}
+
+	r, err := strconv.Atoi(expressionID)
+
+	if err := d.Set("expression_id", r); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("name", expression.Name); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("description", expression.Description); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("regex", expression.Regex); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+//Update the obfuscation expression
+func resourceNewRelicObfuscationExpressionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ProviderConfig).NewClient
+	updateInput := expandObfuscationExpressionUpdateInput(d)
+
+	log.Printf("[INFO] Updating New Relic One obfuscation expression %s", d.Id())
+
+	accountID := selectAccountID(meta.(*ProviderConfig), d)
+	expressionID := d.Id()
+
+	_, err := client.Logconfigurations.LogConfigurationsUpdateObfuscationExpressionWithContext(ctx, accountID, updateInput)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(expressionID)
```

Comment:
No need to setId again on update unless the ID is changing

Reply:
Ok. Made the change